### ISSUE TITLE
Allow settings to be used from non Analyzer context

### ISSUE THREAD

**Issue by hvanbakel** [DotNetAnalyzers/StyleCopAnalyzers#1336]
The current extension method signature
```CSharp
internal static StyleCopSettings GetStyleCopSettings(this SyntaxTreeAnalysisContext context)
```
Is unable to be used from a codefix provider as there is no SyntaxTreeAnalysisContext available.

**Comment by hvanbakel** [comment]
Will try to fix this myself

**Comment by sharwell** [comment]
This is fixed in #1278.
